Nintroller/Nintroller.cs

```diff
@@ -613,6 +669,42 @@ private void ParseReport(byte[] report)
                         case StatusType.Unknown:
                         case StatusType.DiscoverExtension:
                         default:
+                            /* Based on observations, while the Nyko Frontman is connected to the Wiimote (before being configured properly by WiitarThing),
+                               every few seconds the Wiimote sends status reports where the extension controller bit is alternating between 0 and 1
+                               from one report to another. These contradicting reports disrupt the extension discovery process, causing it to toggle between
+                               a Wiimote and a Guitar, all of this while the guitar remains physically connected.
+                               They also disrupt further initialization steps (e.g. a status report claiming no extension is connected arrives while setting up guitar
+                               encryption). To summarize, it's best to ignore a status report in two cases:
+                               1. It's not the report that is specifically requested right after connecting the device (determined by _statusType == StatusType.Unknown,
+                                  hence why _statusType is first initialized to StatusType.DiscoverExtension), and it's received when the current type
+                                  is not known (determined by _state == null, _currentType is initialized to Wiimote in the constructor so it couldn't be used).
+                               2. It's received during guitar encryption setup or IR camera setup (detectable via _ackType).
+                            */
+                            if ((_statusType == StatusType.Unknown && _state == null) ||
+                                (_ackType.ToString().StartsWith("EncryptionSetup") || _ackType.ToString().StartsWith("IR_Step")))
```

Comment:
Would prefer checking against the enum values directly instead of using ToString, as it'll be (at least marginally) faster and avoid allocation garbage every time a report is received in this state.